packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,600 @@
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => set(this.context, 'foo', 'BORK'));
+
+    this.assertText('BORK BORK');
+  }
+
+  ['@test should be able to output a property without binding']() {
+    this.render(`<div id="first">{{unbound content.anUnboundString}}</div>`, {
+      content: {
+        anUnboundString: 'No spans here, son.'
+      }
+    });
+
+    this.assertText('No spans here, son.');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('No spans here, son.');
+
+    this.runTask(() => set(this.context, 'content.anUnboundString', 'HEY'));
+
+    this.assertText('No spans here, son.');
+
+    this.runTask(() => set(this.context, 'content', {
+      anUnboundString: 'No spans here, son.'
+    }));
+
+    this.assertText('No spans here, son.');
+  }
+
+  ['@test should be able to use unbound helper in #each helper']() {
+    this.render(`<ul>{{#each items as |item|}}<li>{{unbound item}}</li>{{/each}}</ul>`, {
+      items: emberA(['a', 'b', 'c', 1, 2, 3])
+    });
+
+    this.assertText('abc123');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('abc123');
+  }
+
+  ['@test should be able to use unbound helper in #each helper (with objects)']() {
+    this.render(`<ul>{{#each items as |item|}}<li>{{unbound item.wham}}</li>{{/each}}</ul>`, {
+      items: emberA([{ wham: 'bam' }, { wham: 1 }])
+    });
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.context.items.setEach('wham', 'HEY'));
+
+    this.assertText('bam1');
+
+    this.runTask(() => set(this.context, 'items',  emberA([{ wham: 'bam' }, { wham: 1 }])));
+
+    this.assertText('bam1');
+  }
+
+  ['@test it should assert unbound cannot be called with multiple arguments']() {
+    let willThrow = () => {
+      this.render(`{{unbound foo bar}}`, {
+        foo: 'BORK',
+        bar: 'BLOOP'
+      });
+    };
+
+    expectAssertion(willThrow, /unbound helper cannot be called with multiple params or hash params/);
+  }
+
+  ['@test should render on attributes']() {
+    this.render(`<a href="{{unbound foo}}"></a>`, {
+      foo: 'BORK'
+    });
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => this.rerender());
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'OOF'));
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'BORK'));
+
+    this.assertHTML('<a href="BORK"></a>');
+  }
+
+  ['@htmlbars syntax form updates on parent re-render']() {
+    let fooBarInstance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        fooBarInstance = this;
+      },
+      foo: 'BORK',
+      barBinding: 'foo'
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: `{{unbound foo}} {{unbound bar}}`
+    });
```

Comment:
Do we actually need a component here? It seems like what we are really testing is that when the user explicitly call `component.rerender()` (which is what `this.rerender()` does), we should re-evaluate the `{{unbound}}`s nested inside it. Seems like the usual `this.render(`{{unbound foo}}`, { foo: ... })` would be just fine. (Which means we can just merge all the with/without parent re-render variants into one? Basically whenever you make changes, it shouldn't update. But when you add a `this.rerender()` it should.)

(By the way, the `barBinding` doesn't need to be specifically tested here. It's just part of the object model and neither htmlbars or glimmer 2 do anything special about them.)